zerver/views/realm.py

```diff
@@ -67,6 +67,11 @@ def update_realm(
     if signup_notifications_stream_id is not None and settings.NEW_USER_BOT is None:
         return json_error(_("NEW_USER_BOT must configured first."))
 
+    # Additional validation of permissions values to add new bot
+    bots_allowed_for_non_admins_types = [Realm.ALL, Realm.NO_GENERIC_BOTS, Realm.NONE]
```

Comment:
I would probably put this list in `class Realm` as `Realm.BOT_CREATION_POLICY_TYPES`, so we don't need this piece of code to change if we later add another type.